File: alice/paskills/dialogovo/src/main/java/ru/yandex/alice/paskill/dialogovo/utils/AnythingFromStringJacksonSerializer.java
package ru.yandex.alice.paskill.dialogovo.utils;

import java.io.IOException;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializerProvider;

public class AnythingFromStringJacksonSerializer extends JsonSerializer<String> {

    @Override
    public void serialize(String value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
        var mapper = new ObjectMapper();
        var tree = mapper.readTree(value);

        gen.writeObject(tree);
    }
}

File: alice/kronstadt/core/src/main/java/utils/AnythingFromStringJacksonSerializer.kt
package ru.yandex.alice.kronstadt.core.utils

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializerProvider
import java.io.IOException

class AnythingFromStringJacksonSerializer : JsonSerializer<String>() {

    @Throws(IOException::class)
    override fun serialize(value: String, gen: JsonGenerator, serializers: SerializerProvider) {
        val mapper = ObjectMapper()
        val tree = mapper.readTree(value)
        gen.writeObject(tree)
    }
}
